tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/WritePackageUsageData.cs

```diff
@@ -180,28 +180,44 @@ public override bool Execute()
                 assetFiles,
                 assetFile =>
                 {
-                    var properties = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    JObject jObj;
 
                     using (var file = File.OpenRead(Path.Combine(RootDir, assetFile)))
                     using (var reader = new StreamReader(file))
                     using (var jsonReader = new JsonTextReader(reader))
                     {
-                        while (jsonReader.Read())
-                        {
-                            if (jsonReader.TokenType == JsonToken.PropertyName &&
-                                jsonReader.Value is string value)
-                            {
-                                properties.Add(value);
-                            }
-                        }
+                        jObj = (JObject)JToken.ReadFrom(jsonReader);
                     }
 
+                    var properties = new HashSet<string>(
+                        jObj.SelectTokens("$.targets.*")?.Children()
```

Comment:
When does `SelectTokens` return null? The `?.` looks like it'll apply incorrectly and make the whole expression `null` when that happens. In a local repro attempt, it returns empty if there aren't any targets elements.

Reply:
Agreed.  It doesn't look like SelectTokens() returns null.  I'll remove the ?